Sources/MacroToolkit/EnumCase.swift

```diff
@@ -4,27 +4,41 @@ import SwiftSyntax
 public struct EnumCase {
     public var _syntax: EnumCaseElementSyntax
 
-    public init(_ syntax: EnumCaseElementSyntax) {
+    public init(_ syntax: EnumCaseElementSyntax, rawRepresentableType: EnumRawRepresentableType? = nil, precedingCase: EnumCase? = nil) {
         _syntax = syntax
+        value = {
```

Comment:
Refactor into a private static helper function to avoid the immediately-invoked closure pattern (I prefer not to use it). The helper could be called `inferRawValue` or something.

Reply:
avoided the closure pattern using if and switch expressions